src/types/redis_stream.cc

```diff
@@ -737,6 +774,73 @@ rocksdb::Status Stream::GetStreamInfo(const rocksdb::Slice &stream_name, bool fu
   return rocksdb::Status::OK();
 }
 
+rocksdb::Status Stream::GetGroupInfo(const Slice &stream_name,
+                                     std::vector<std::pair<std::string, StreamConsumerGroupMetadata>> &group_metadata) {
+  std::string ns_key = AppendNamespacePrefix(stream_name);
```

Comment:
I think we don't need to lock the key? because it's good to read the snapshot only.